spec/models/breakeven_table_question_spec.rb

```diff
@@ -0,0 +1,189 @@
+            { quantity: 75.0, total: 9_000.0 },
+            { quantity: 150.0, total: 18_000.0 },
+            { quantity: 225.0, total: 27_000.0 },
+            { quantity: 300.0, total: 36_000.0 },
+          ] },
+          { name: 'Product 3', quantity: 100.0, amount: 150.0, total: 15_000.0, rampup: [
+            { quantity: 25.0, total: 3_750.0 },
+            { quantity: 50.0, total: 7_500.0 },
+            { quantity: 75.0, total: 11_250.0 },
+            { quantity: 100.0, total: 15_000.0 },
+          ] },
+        ],
+        total_revenue: 131_000,
+        total_revenue_rampup: [3_2750, 65_500, 98_250, 131_000],
+        cogs: [
+          { name: 'Product 1', quantity: 800.0, amount: 50.0, total: 40_000.0, rampup: [
+            { quantity: 200.0, total: 10_000.0 },
+            { quantity: 400.0, total: 20_000.0 },
+            { quantity: 600.0, total: 30_000.0 },
+            { quantity: 800.0, total: 40_000.0 },
+          ] },
+          { name: 'Product 2', quantity: 300.0, amount: 60.0, total: 18_000.0, rampup: [
+            { quantity: 75.0, total: 4_500.0 },
+            { quantity: 150.0, total: 9_000.0 },
+            { quantity: 225.0, total: 13_500.0 },
+            { quantity: 300.0, total: 18_000.0 },
+          ] },
+          { name: 'Product 3', quantity: 100.0, amount: 70.0, total: 7_000.0, rampup: [
+            { quantity: 25.0, total: 1_750.0 },
+            { quantity: 50.0, total: 3_500.0 },
+            { quantity: 75.0, total: 5_250.0 },
+            { quantity: 100.0, total: 7_000.0 },
+          ] },
+        ],
+        total_cogs: 65_000,
+        total_cogs_rampup: [16_250, 32_500, 48_750, 65_000],
+        gross_margin: 66_000,
+        gross_margin_rampup: [16_500, 33_000, 49_500, 66_000],
+        fixed_costs: [
+          { name: 'Rent', amount: 15_000.0 },
+          { name: 'Worker owners', amount: 28_000.0 },
+          { name: 'Employees', amount: 10_000.0 },
+          { name: 'Sales', amount: 10_000 },
+          { name: 'Utilities', amount: 2_000.0 },
+          { name: 'Insurance', amount: 1_000.0 },
+        ],
+        total_fixed_costs: 66_000,
+        total_fixed_costs_rampup: [66_000, 66_000, 66_000, 66_000],
+        net_margin: 0,
+        net_margin_rampup: [-49_500, -33_000, -16_500, 0],
+        periods: 4,
+        units: 'Months'
+      }
+    end
+
+    subject { BreakevenTableQuestion.new(json) }
+
+    it 'calculates entire report properly' do
+      expect(subject.report).to eq results
+    end
+
+    [
+      :revenue,
+      :total_revenue,
+      :total_revenue_rampup,
+      :cogs,
+      :total_cogs,
+      :total_cogs_rampup,
+      :gross_margin,
+      :gross_margin_rampup,
+      :fixed_costs,
+      :total_fixed_costs,
+      :total_fixed_costs_rampup,
+      :net_margin,
+      :net_margin_rampup,
+      :periods,
+      :units,
+    ].each do |row|
+      it "calculates #{row}" do
```

Comment:
I was finding it hard to track down errors, without more specific tests. I left this one to ensure extra fields didn't sneak into the results. I am thinking of a better way I could do this.